pkg/util/encryption/xchacha20poly1305.go

```diff
@@ -26,8 +26,8 @@ type aeadCipher struct {
 	randRead func([]byte) (int, error)
 }
 
-func NewXChaCha20Poly1305(ctx context.Context, _env env.Interface) (Cipher, error) {
-	key, err := _env.GetSecret(ctx, env.EncryptionSecretName)
+func NewXChaCha20Poly1305(ctx context.Context, _env env.Interface, secretName string) (Cipher, error) {
+	key, err := _env.GetSecret(ctx, secretName)
```

Comment:
I don't like that we have dependency on `env.Interface` here. Wondering if it makes sense to do `_env.GetSecret(ctx, secretName)` in the calling code.